src/software/ai/hl/stp/play/enemy_free_kick_play_test.cpp

```diff
@@ -11,56 +11,73 @@
-            // chosen to be generally in the way of the the front 2 enemy robots and the
-            // enemy robot performing the free kick, based on where the enemy robots are
-            // initialized in the test.
-            Rectangle robot_four_shadowing_rect(Point(-1, 0.25), Point(-0.75, -0.25));
-            Rectangle robot_five_shadowing_rect(Point(-2, -1), Point(-1.75, -1.25));
-            robotInPolygon(4, robot_four_shadowing_rect, world_ptr, yield);
-            robotInPolygon(5, robot_five_shadowing_rect, world_ptr, yield);
-
-            // One Friendly robot defending the exterior of defense box to the right of
-            // the goalie
-            Point goalie_position = world_ptr->friendlyTeam().goalie()->position();
-            Rectangle crease_defender_rect(
-                goalie_position,
-                Point(goalie_position.x() + 0.2, goalie_position.y() - 0.2));
-            robotInPolygon(3, crease_defender_rect, world_ptr, yield);
+        [num_enemy_robots](std::shared_ptr<World> world_ptr,
+                           ValidationCoroutine::push_type& yield) {
+            if (num_enemy_robots == 6)
+            {
+                // Wait for all robots to come to a halt
+                robotHalt(world_ptr, yield);
+
+                // Two robots defending close to the enemy robot performing the free kick
+                Rectangle shadowing_free_kicker_rect(Point(0, 3), Point(1, 2));
+                robotInPolygon(1, shadowing_free_kicker_rect, world_ptr, yield);
+                robotInPolygon(3, shadowing_free_kicker_rect, world_ptr, yield);
+
+                // Two friendly robots in position to shadow enemy robots. Rectangles are
+                // chosen to be generally in the way of the the front 2 enemy robots and
+                // the enemy robot performing the free kick, based on where the enemy
+                // robots are initialized in the test.
+                Rectangle robot_four_shadowing_rect(Point(-1, 0.25), Point(-0.75, -0.25));
+                Rectangle robot_five_shadowing_rect(Point(-2, -1), Point(-1.75, -1.25));
+                robotInPolygon(4, robot_four_shadowing_rect, world_ptr, yield);
+                robotInPolygon(5, robot_five_shadowing_rect, world_ptr, yield);
+
+                // One Friendly robot defending the exterior of defense box to the right
+                // of the goalie
+                Point goalie_position = world_ptr->friendlyTeam().goalie()->position();
+                Rectangle crease_defender_rect(
+                    goalie_position,
+                    Point(goalie_position.x() + 0.2, goalie_position.y() + 0.4));
+                robotInPolygon(2, crease_defender_rect, world_ptr, yield);
+            }
+            else
+            {
+                while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(5))
+                {
+                    yield("Timestamp not at 5s");
+                }
+            }
```

Comment:
for the 5,4,3,2,1 cases, why do we just wait for 5 seconds?

Reply:
checking for no segfault :) thoughts?

Reply:
ah gotcha, i think a comment here would be nice

maybe it can also be another test rather than an else block in a test, but thats a nit, comment would be good though

Reply:
fixed